hardware_interface/include/hardware_interface/robot_hw.h

```diff
@@ -80,6 +107,7 @@ class RobotHW : public InterfaceManager
    */
   virtual bool init(ros::NodeHandle& /*root_nh*/, ros::NodeHandle &/*robot_hw_nh*/) {return true;}
```

Comment:
Is there a reason why all parameters are commented in the headers? I know only the type is required in the declaration of a method but I am asking because doxygen gave me warnings of the missing parameter names.

Reply:
That's... A good question. I suspect that's a mistake. Commenting out unused parameters at the function definition is sometimes done to suppress compiler warnings, but I don't think there's any reason to comment them out in the declaration like this. Would you mind un-commenting them here in the header, and double-checking that this satisfies both Doxygen and the compiler? (ie there are no compiler warnings when you compile with `-Wunused-parameter`/`-Wunused`/`-Wextra`)

Reply:
See #399. Looks like these were probably extra accidental changes.

Reply:
Thanks for the link. I've uncommented the parameters and compiled the `hardware_interface` tests (where `robot_hw.h` gets included) using: `catkin build -v --pre-clean --cmake-args -DCMAKE_C_FLAGS="-Wall -Wextra -Wunused-parameter -Wunused" --catkin-make-args tests` without any warnings. Should I commit these changes (uncommenting parameters) to this PR or in a separate one?

Reply:
Let's split it out, this is already going to be a fairly lengthy PR. Thanks :)